src/binaryutils.cpp

```diff
@@ -89,79 +89,25 @@ Status convertPrecision(const cv::Mat& src, cv::Mat& dst, const ovms::Precision
 }
 
 Status validateLayout(const std::shared_ptr<TensorInfo>& tensorInfo) {
-    if ((tensorInfo->getLayout() != "NHWC") &&
-        (tensorInfo->getLayout() != Layout::getUnspecifiedLayout()) &&  // handle DAG
-        (tensorInfo->getLayout() != Layout::getDefaultLayout())) {      // handle model without Layout set
+    static const std::string binarySupportedLayout = "N...HWC";
+    if (!tensorInfo->getLayout().createIntersection(Layout(binarySupportedLayout), tensorInfo->getShape().size()).has_value()) {
+        SPDLOG_ERROR("Endpoint needs to be compatible with {} to support binary image inputs, actual: {}",
```

Comment:
Shouldn;t this be DEBUG? We could spam OVMS logs with predict requests otherwise

Reply:
There are plenty of non-debug logs when request is compatible with endpoint.
Examples when precision or shape is wrong:

```
[2022-02-14 12:34:50.702][832][serving][warning][modelinstance.cpp:957] Model: resnet, version: 1 validation of inferRequest failed. Status Code: 61, Error: Invalid input precision - Expected: FP32; Actual: FP64; input name: 0
[2022-02-14 12:35:09.896][833][serving][warning][modelinstance.cpp:957] Model: resnet, version: 1 validation of inferRequest failed. Status Code: 60, Error: Invalid input shape - Expected: (1,3,224,224); Actual: (1,4,224,224); input name: 0
```

If we want to move all errors to DEBUG, this should be separate task. @atobiszei